source/d3d11/depth_counter_tracker.cpp

```diff
@@ -0,0 +1,135 @@
+#include "depth_counter_tracker.hpp"
+#include "com_ptr.hpp"
+#include <math.h>
+
+depth_counter_tracker::depth_counter_tracker() : 
+	_drawcalls(0), 
+	_vertices(0) 
+{ }
+
+depth_counter_tracker::~depth_counter_tracker() { }
+
+void depth_counter_tracker::track_depthstencil(ID3D11DepthStencilView* to_track)
+{
+	if (nullptr != to_track)
+	{
+		if (_counters_per_used_depthstencil.find(to_track) == _counters_per_used_depthstencil.end())
+		{
+			const depthstencil_counter_info counters = {};
+			to_track->AddRef();
+			_counters_per_used_depthstencil.emplace(to_track, counters);
+		}
+	}
+}
+
+void depth_counter_tracker::log_drawcalls(ID3D11DepthStencilView* depthstencil, UINT drawcalls, UINT vertices)
+{
+	if (nullptr == depthstencil || drawcalls==0)
```

Comment:
Maybe make this an `assert` instead, since the same branch already exists in `D3D11DeviceContext::log_drawcall_stats_in_tracker`?

Reply:
As this function is only called from `log_drawcall-stats_in_tracker` it's indeed redundant.